feat: @contentful/f36-popover (#1204)
contentful/forma-36: packages/components/popover/Popover.mdx

---
title: 'Popover'
type: 'component'
status: 'stable'
slug: /components/popover/
github: 'https://github.com/contentful/forma-36/tree/master/packages/components/popover'
storybook: 'https://f36-storybook.contentful.com/?path=/story/components-popover--basic'
typescript: ./src/Popover.tsx

> ---

Reviewer: Shouldn't we also render the props table?

Popover is used to display some content on top of another, and should be paired with a clickable trigger element.

> It is a base for other more specific component, like `Menu`, `Autocomplete` and `Multiselect`. Please, consider using these specific componenets to cover your needs.

Author: We (with designers) decided to give the users more flexibility. Since we are removing `Dropdown` component which was something  in the middle between raw popover and specific components like `Menu` or `Multiselect`.

> I guess we have other components like this, right?
What do you mean?

> It is a base for other more specific component, like `Menu`, `Autocomplete` and `Multiselect`. Please, consider using these specific componenets to cover your needs.

Reviewer 2: Nevermind! hahahaha thanks for clarifying 👌

## Table of contents

- [How to use Popover](#how-to-use-Popover)
- [Code examples](#code-examples)
- [Content recommendations](#content-recommendations)
- [Accessibility](#accessibility)

## How to use Popover

- If `Menu`, `Autocomplete` and `Multiselect` components are not covering your use cases, only then you can use `Popover`.
- Double check with designers before using this component. Probably you can just use one of the more specific components, like `Menu`, `Autocomplete` and `Multiselect`
- Keep in mind that everything related to accessibility for the popover content you have to develop on your own.

## Code examples

- Pass trigger component as a child for `PopoverTrigger`.
- Pass popover content as a child for `PopoverContent`

```jsx
<React.Fragment>
  <TextLink variant="primary">Text link</TextLink>

> </React.Fragment>

Author: Why the `TextLink`? ahah 😄 
I just forgot to update examples after copying it from `TextLink` 🤦‍♂️

---
title: 'Popover'
type: 'component'
status: 'stable'
slug: /components/popover/
github: 'https://github.com/contentful/forma-36/tree/master/packages/components/popover'
storybook: 'https://f36-storybook.contentful.com/?path=/story/components-popover--basic'
typescript: ./src/Popover.tsx
---

Popover is used to display some content on top of another, and should be paired with a clickable trigger element.
It is a base for other more specific component, like `Menu`, `Autocomplete` and `Multiselect`. Please, consider using these specific componenets to cover your needs.

## Table of contents

- [How to use Popover](#how-to-use-Popover)
- [Code examples](#code-examples)
- [Content recommendations](#content-recommendations)
- [Accessibility](#accessibility)

## How to use Popover

> - If `Menu`, `Autocomplete` and `Multiselect` components are not covering your use cases, only then you can use `Popover`.

Reviewer 3: ```suggestion
- Only If `Menu`, `Autocomplete` and `Multiselect` components are not covering your use cases, you should use `Popover`.
```

> - Double check with designers before using this component. Probably you can just use one of the more specific components, like `Menu`, `Autocomplete` and `Multiselect`

Reviewer 3: ```suggestion
- Before using this component, double-check your design requirements. We are providing `Menu`, `Autocomplete`, and `Multiselect` for more specific use-cases and they can address your needs.
```

> - Keep in mind that everything related to accessibility for the popover content you have to develop on your own.

Reviewer 3: ```suggestion
- Keep in mind that you will have to implement everything related to accessibility for the popover content.
```
